data/lua.js

```diff
@@ -136,107 +136,66 @@ are not restricted.`
-				text: `Maximum number of attempts for a request.`
 			},
 			max_auto_retry_delay: {
+				dovecot_setting: 'http_client_max_auto_retry_delay',
 				hash_arg: true,
-				type: 'int',
-				text: `
-Maximum acceptable delay in seconds for automatically retrying/redirecting
-requests.
-
-If a server sends a response with a \`Retry-After\` header that causes a
-delay longer than this, the request is not automatically retried and
-the response is returned.`
 			},
 			max_connect_attempts: {
+				dovecot_setting: 'http_client_max_connect_attempts',
 				hash_arg: true,
-				type: 'int',
-				text: `
-Maximum number of connection attempts to a host before all associated
-requests fail.
-
-If > 1, the maximum will be enforced across all IPs for that host, meaning
-that IPs may be tried more than once eventually if the number of IPs is
-smaller than the specified maximum attempts. If the number of IPs is
-higher than the maximum attempts, not all IPs are tried.
-
-If <= 1, all IPs are tried at most once.`
 			},
 			max_idle_time: {
+				dovecot_setting: 'http_client_max_idle_time',
 				hash_arg: true,
-				type: 'int',
-				text: `
-Maximum time a connection will idle before disconnecting.
-
-If parallel connections are idle, the duplicates will end earlier based on
-how many idle connections exist to that same service.`
 			},
 			max_redirects: {
-				default: '0; redirects refused',
+				dovecot_setting: 'http_client_request_max_redirects',
 				hash_arg: true,
-				type: 'int',
-				text: `Maximum number of redirects for a request.`
 			},
 			proxy_url: {
+				dovecot_setting: 'http_client_proxy_url',
 				hash_arg: true,
-				type: 'string',
 				text: `Proxy URL to use, can include username and password.`
 			},
 			request_absolute_timeout: {
-				default: '0; no timeout',
+				dovecot_setting: 'http_client_request_absolute_timeout',
 				hash_arg: true,
-				type: 'int',
-				text: `Max total time to wait for HTTP request to finish, including retries and everything else.`
 			},
 			request_timeout: {
-				default: '60000 msec',
+				dovecot_setting: 'http_client_request_timeout',
 				hash_arg: true,
-				type: 'int',
-				text: `Max time to wait for HTTP response before retrying.`
 			},
 			soft_connect_timeout: {
-				default: '0; wait until current connection attempt finishes',
+				dovecot_setting: 'http_client_soft_connect_timeout',
 				hash_arg: true,
-				type: 'int',
-				text: `Time to wait for \`connect()\` (and SSL handshake) to finish for the first connection before trying the next IP in parallel.`
 			},
 			rawlog_dir: {
+				dovecot_setting: 'http_client_rawlog_dir',
 				hash_arg: true,
-				type: 'string',
-				text: `Directory for writing raw log data for debugging purposes. Must be writable by the process creating this log.`
 			},
 			user_agent: {
-				default: '<none>',
+				dovecot_setting: 'http_client_user_agent',
 				hash_arg: true,
-				type: 'string',
-				text: `\`User-Agent:\` header.`
 			},
```

Comment:
What about SSL settings?

Reply:
hm, yeah, have to link those too.

Reply:
Also kind of annoying having to specify each setting separately. Would be easier if there was just a way to get everything with http-client tag.

Reply:
We're definitely going to forget to update these settings when we add new http-client or ssl settings..

Reply:
The problem with using tags is that
- they don't (necessarily) matchup between lua settings and core settings . Thus, more dynamic code manipulation needed. Data files were designed to be JSON-only so that developers don't need to know javascript
- Automatically adding a setting if added to a tag in settings.js is a bug, not a feature.  You still need to expose that particular setting in Lua code.  So there's still a manual process needed to implement these new settings, and I don't think it's asking much to have to review/update Lua documentation if/when a new setting is exposed.

The current solution seems like a valid compromise of making it easy to add an additional feature using straightforward JSON, and making it very clear in the lua.js file itself what the actual settings are.  There's not THAT many http-client or ssl settings that we need to overly automate this process.

Reply:
actually all settings are automatically exposed in lua code as it passes them along to settings parser.

Reply:
Right, that's why I'm thinking it's going to be an issue in the future. We add a new http-client or ssl-client setting without thinking of Lua at all. The new setting will be exposed to Lua without any extra code, but documentation is forgotten to be updated.

In theory we could have exposed all settings to Lua, not just http-client and ssl settings. But then we couldn't have given errors about unknown settings. And I'm not sure if there are any other settings that may be useful. Perhaps lib-dns once it's done, but likely not worth it.

Reply:
can we merge this and create new ticket for fixing this?